recipes/lua/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+import os
+from distutils.dir_util import copy_tree
+
+
+license = """================================================================================
+Copyright © 1994–2019 Lua.org, PUC-Rio.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+class LuaConan(ConanFile):
+    no_copy_source = True
+
+    name = "lua"
+    description = "Lua is a powerful, efficient, lightweight, embeddable scripting language."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.lua.org/"
+    topics = ("conan", "lua", "scripting")
+    license = "MIT"
+    generators = "cmake"
+    settings = "os", "compiler", "arch", "build_type"
+    exports_sources = ["CMakeLists.txt"]
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    options = {"shared": [False, True], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "lua-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["SOURCE_SUBDIR"] = self._source_subfolder
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        os.makedirs(os.path.join(self.package_folder, "licenses"))
+        tools.save(os.path.join(self.package_folder, "licenses", "COPYING.txt"), license)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs.append("include/lua")
```

Comment:
```suggestion
        self.cpp_info.includedirs.append("include/lua")
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

Take a look in the makefile